allennlp/training/learning_rate_schedulers/linear_with_warmup.py

```diff
@@ -0,0 +1,46 @@
+from overrides import overrides
+import torch
+
+from allennlp.training.learning_rate_schedulers.learning_rate_scheduler import LearningRateScheduler
+
+
+@LearningRateScheduler.register("linear_with_warmup")
```

Comment:
(I saw this by coincidence during the tech orientation)

Why do we need this class? It seems to be exactly the same as `PolynomialDecay` with `end_learning_rate=0.0` and `power=`1.0`. Perhaps we should fix up `PolynomialDecay` to take in `num_epochs` and `num_steps_per_epoch` instead of `total_steps` and then get rid of `LinearWithWarmup`?

Reply:
I want to keep it as a class, because it's easier to remember when you're writing a config, but `LinearWithWarmup` could just be a sub-class of `PolynomialDecay`, and pass in the right parameters.